### PR TITLE
PodFailing scenario: updated assessment_runner.go

### PR DESCRIPTION
**Pull request by chathuryaadapa**
Added code to handle the pod fail scenario by cross checking the error message we pass in .WithFailReason(reason string)

Fixes: https://github.com/confidential-containers/cloud-api-adaptor/issues/1760

### CODE REVIEW

File: src/cloud-api-adaptor/test/e2e/assessment_runner.go [hunk 1]
```diff
@@ -416,6 +423,25 @@ func (tc *TestCase) Run() {
...
+						}
+						if strings.Contains(string(jsonData), tc.FailReason) {
+							log.Infof("failed due to expected reason %s", tc.FailReason)
+						} else {
+							t.Fatal(err)
+						}
+					} else {
+						log.Infof("Pod Failed If you want to cross check please give .WithFailReason(error string)")
+					}
+					tc.assert.HasPodVM(t, tc.pod.Name)
```
Comment (genjuro214) [inline src/cloud-api-adaptor/test/e2e/assessment_runner.go:442]: If pod is not running, the Pod VM may be not created in some case. Is it necessary to check Pod VM existence here?

**Comment by stevenhorsman** [comment]
The commit message check is failing as your commit header should start with `tests/e2e` and be 50 chars long.
I think you can also squash the commits together as they are all tightly related.
Let me know if you hit any problems with this. Thanks!

**Review by stevenhorsman — APPROVED** [review]
I've not tested it locally, and don't know the specific cases that failed, but I ran the libvirt e2e tests and it passes still and the code looks ok

**Review by liudalibj — APPROVED** [review]
LGTM

**Comment by liudalibj** [comment]
@chathuryaadapa would you like squash the commits together? after that we are ready to merge this PR.

**Pull request merged**